src/test/kotlin/org/jetbrains/research/transformations/util/ParserSetup.kt

```diff
@@ -1,33 +1,91 @@
+    private const val INVERSE_PARSER_RELATIVE_PATH =
+        "$REPOSITORY_ROOT_FOLDER/src/main/python/inverse_parser/inverse_parser_3.py"
+
     private const val PARSER_NAME = "pythonparser"
-    private val TARGET_PATH = "${getTmpPath()}/$PARSER_NAME"
+
+    private val TARGET_PARSER_PATH = "${getTmpPath()}/$PARSER_NAME"
+    private val TARGET_INVERSE_PARSER_PATH = "${getParserRepositoryPath()}/$INVERSE_PARSER_RELATIVE_PATH"
 
     /**
-     * Puts parser file to the target path.
-     * @param targetPath - path to the final file destination,
-     * where it will be putted.
+     * Runs inverse_parser_3 using python3, which it gets from [PYTHON3_PROPERTY] or sets the default one
+     * To set [PYTHON3_PROPERTY], please, add -P[PYTHON3_PROPERTY]=/path/to/python3 to the command line
      */
-    private fun putParserToTargetPath(targetPath: String= TARGET_PATH) {
+    fun getCommandForInverseParser(XMLPath: String): Util.Command {
+        val defaultPythonBin = if (SystemUtils.IS_OS_WINDOWS) "where python3" else "which python3"
+        val pythonBin = System.getProperty(PYTHON3_PROPERTY)?.let { "echo $it" } ?: defaultPythonBin
+        return Util.Command(
+            listOf("/bin/bash", "-c", "$($pythonBin) $TARGET_INVERSE_PARSER_PATH $XMLPath"),
+            environment = mapOf("PYTHONPATH" to getRepositoryRootPath())
+        )
+    }
+
+    private fun getRepositoryRootPath() =
+        "${getParserRepositoryPath()}/$REPOSITORY_ROOT_FOLDER"
 
-        try {
-            LOG.info("Putting parser into $targetPath")
-            val pythonparserFile = File(javaClass.getResource("$PARSER_NAME.py").path)
-            pythonparserFile.copyTo(File(targetPath))
+    /**
+     * Get parser repository path in this project in the resources folder
+     */
+    private fun getParserRepositoryPath(): String {
+        val zipFilePath = Paths.get(javaClass.getResource(PARSER_ZIP_NAME).path)
+        // TODO: find a better way for it
+        return zipFilePath.parent.toString().replace("build/resources/test/", "src/test/resources/")
+    }
 
-        } catch (e: FileAlreadyExistsException) {
-            LOG.info("Parser file is already in $TARGET_PATH")
+    /**
+     * Update parser repository
+     * @param toUpdate - it is true, if a new version of the parser needs to download
+     */
+    private fun updateParserRepo(toUpdate: Boolean) {
+        val zipFilePath = Paths.get("${getParserRepositoryPath()}/$PARSER_ZIP_NAME")
+        if (toUpdate) {
+            LOG.info("Updating the current master zip")
+            val file: InputStream = URL(PARSER_REPOSITORY_ZIP_URL).openStream()
+            Files.copy(file, zipFilePath, StandardCopyOption.REPLACE_EXISTING)
         }
+        unzipParserRepo()
+    }
 
+    private fun unzipParserRepo(
+        zipParserRepoPath: String
+        = Paths.get("${getParserRepositoryPath()}/$PARSER_ZIP_NAME").toString()
```

Comment:
Please, let's move it to the separate method, it's hard to read it inside the arguments' block